src/lib.rs

```diff
@@ -973,3 +984,35 @@ fn getenv_unwrap(v: &str) -> String {
 fn fail(s: &str) -> ! {
     panic!("\n{}\n\nbuild script failed, must exit now", s)
 }
+
+// Returns the major, minor and patch versions of cmake
+fn cmake_version(executable: &OsStr) -> Option<(u8, u8, u8)> {
+    let cmd = Command::new(executable).arg("--version").output().ok()?;
+    let version = String::from_utf8_lossy(&cmd.stdout);
+    let version: Vec<&str> = version.split_whitespace().collect();
+    let version: Vec<&str> = version[2].split('.').collect();
+    let mut temp = vec![];
+    for i in version.iter() {
+        let curr = i.parse().ok()?;
+        temp.push(curr);
+    }
+    Some((temp[0], temp[1], temp[2]))
+}
+
+// Check if cmake has the --parallel flag by checking against version 3.12
+fn cmake_has_parallel(executable: &OsStr) -> bool {
+    if let Some(cmake_version) = cmake_version(&executable) {
+        if cmake_version >= (3, 12, 0) {
+            true
+        } else {
+            println!(
+                "cargo:warning=CMake version {}.{}.{} detected, No parallel flag will be passed!",
+                cmake_version.0, cmake_version.1, cmake_version.2
+            );
+            false
+        }
+    } else {
+        println!("cargo:warning=cmake-rs might have failed at parsing the version output!");
```

Comment:
I think this warning is a bit misleading, this crate did indeed fail to parse the output instead of "might" have failed to parse, and I think it would be good to mention the output that failed to parse to assist with debugging if necessary.